pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -371,5 +442,50 @@ int64_t     *numpy_to_ndarray_shape(int64_t *np_shape, int nd)
     for (int i = 0; i < nd; i++)
         nd_shape[i] = np_shape[i];
     return nd_shape;
+}
 
+/*
+** takes an array containing the shape of an array 'shape', number of a certain dimension 'nd', and the number of the array's dimensions
+** returns the stride (number of elements to jump in a dimension) of the 'nd`th dimension
+*/
+
+int get_dimension_stride(int64_t *shape, int nd, int max_nd)
+{
+    int product = 1;
+
+    for (int i = nd; i < max_nd; ++i)
+        product *= shape[i];
+    return (product);
+}
+
+int element_index(t_ndarray arr, uint32_t element_num, int nd)
+{
+    if (arr.order == order_c)
+        return element_num;
+    if (nd == 0)
+        return (0);
+    if (nd == arr.nd)
+        return (element_num % arr.shape[nd - 1]) * arr.strides[nd - 1] + element_index(arr, element_num, nd - 1);
+    int true_index = (element_num / (get_dimension_stride(arr.shape, nd, arr.nd)));
+    if (true_index >= arr.shape[nd - 1])
+        true_index = true_index % arr.shape[nd - 1];
+    return (true_index * arr.strides[nd - 1] + element_index(arr, element_num, nd - 1));
+}
+
+void array_copy_data(t_ndarray *dest, t_ndarray src)
+{
+    unsigned char *d = (unsigned char*)dest->raw_data;
+    unsigned char *s = (unsigned char*)src.raw_data;
+
+    if (dest->order == order_c && src.order == order_c)

```

Comment:
Not `if dest->order == src.order` ?

Reply:
That will work if we make sure that they are `order_c`, I do think it's more readable, because it kind of says: `they need to be of the same order, for this copy to work`

Reply:
`order_f` won't work if they do not have the same shape

Reply:
I should add an extra condition for `order_f`

Reply:
Yes, I think that we said we would use the numpy convention, we said that we should add docs to make sure we are on the same page

Reply:
I am still not 100% clear when I read them. At the beginning you explain how it works for numpy, but I don't see an equivalent explanation for our code in C or a mention that it works like numpy. It can't be assumed because we do something completely different in Fortran

Reply:
https://github.com/pyccel/pyccel/pull/1128/commits/0de084f0a81657124b0ec3afb9dd38c65ee9af3d